tests/perf-test/SidebarUtils.perf-test.ts

```diff
@@ -24,60 +27,60 @@ afterEach(() => {
     Onyx.clear();
 });
 
-const getMockedReportsMap = (length = 100) => {
-    const mockReports = Array.from({length}, (v, i) => {
-        const reportID = i + 1;
-        const participants = [1, 2];
-        const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;
-        const report = LHNTestUtils.getFakeReportWithPolicy(participants, 1, true);
+const getMockedReports = (length = 500) =>
+    createCollection<Report>(
+        (item) => `report_${item.reportID}`,
+        (index) => createRandomReport(index),
+        length,
+    ) as Partial<Report>;
 
-        return {[reportKey]: report};
-    });
-
-    return Object.assign({}, ...mockReports) as Partial<Report>;
-};
-
-const getMockedPoliciesMap = (length = 100) => {
-    const mockPolicies = Array.from({length}, (v, i) => {
-        const policyID = i + 1;
-        const policyKey = `${ONYXKEYS.COLLECTION.POLICY}${policyID}`;
-        const policy = LHNTestUtils.getFakePolicy();
-
-        return {[policyKey]: policy};
-    });
-
-    return Object.assign({}, ...mockPolicies) as Record<string, Policy>;
-};
+const reportActions = createCollection<ReportAction>(
+    (item) => `${item.reportActionID}`,
+    (index) => createRandomReportAction(index),
+);
 
-const mockedResponseMap = getMockedReportsMap(10000);
+const mockedResponseMap = getMockedReports(10000);
 
 test('getOptionData on 10k reports', async () => {
-    const report = LHNTestUtils.getFakeReportWithPolicy([1, 2], 1, true) as Report;
-    const reportActions = ReportTestUtils.getMockedReportActionsMap();
+    const report = createRandomReport(1);
     const personalDetails = LHNTestUtils.fakePersonalDetails;
     const preferredLocale = 'en';
-    const policy = LHNTestUtils.getFakePolicy() as Policy;
-    const reportAction = ReportTestUtils.getFakeReportAction(1, 'ADDCOMMENT') as unknown as ReportAction;
+    const policy = createRandomPolicy(1);
+    const parentReportAction = createRandomReportAction(1);
 
     Onyx.multiSet({
         ...mockedResponseMap,
     });
 
     await waitForBatchedUpdates();
-    await measureFunction(() => SidebarUtils.getOptionData(report, reportActions, personalDetails, preferredLocale, policy, reportAction), {runs: 20});
+    await measureFunction(() => SidebarUtils.getOptionData(report, reportActions, personalDetails, preferredLocale, policy, parentReportAction), {runs: 20});
 });
 
 test('getOrderedReportIDs on 10k reports', async () => {
     const currentReportId = '1';
-    const allReports = getMockedReportsMap() as Record<string, Report>;
+    const allReports = getMockedReports() as Record<string, Report>;
```

Comment:
is casting still required? type should be properly inferred

Reply:
Added type to the `mockedResponseMap`, so no casting is needed here